### PR TITLE
METRO-22 Subscriber unit test changes

### PR DESCRIPTION
**Pull request by sumitkrzp**
# Description

Added unit test cases to improve code coverage.

# How Has This Been Tested?

Tested by running test cases on local.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have manually tested my code to the best of my abilities.

### CODE REVIEW

File: internal/subscriber/core_test.go
Comment (sumitkrzp) [reply internal/subscriber/core_test.go]: acknowledged the changes

File: internal/subscriber/core_test.go
Comment (sakshi-goyal-razorpay) [inline internal/subscriber/core_test.go:186]: why are we using Equalf and not Equal..?

File: internal/subscriber/core_test.go
Comment (sakshi-goyal-razorpay) [inline internal/subscriber/core_test.go:222]: no need to create a func.. Use NewCore directly wherever needed.

File: internal/subscriber/ordering_test.go
Comment (sakshi-goyal-razorpay) [inline internal/subscriber/ordering_test.go:171]: assert.Zero

File: internal/subscriber/subscriberimpl_test.go
Comment (sakshi-goyal-razorpay) [inline internal/subscriber/subscriberimpl_test.go:256]: why have we changed this..?

File: internal/subscriber/core_test.go
Comment (sakshi-goyal-razorpay) [inline internal/subscriber/core_test.go:191]: why don't use getMockConsumer function

File: internal/subscriber/subscriberimpl_test.go
Comment (sumitkrzp) [reply internal/subscriber/subscriberimpl_test.go:256]: This will increase coverage and test healthMonitoring is working fine or not

File: internal/subscriber/core_test.go
Comment (sumitkrzp) [reply internal/subscriber/core_test.go:186]: I though EqualF will print detailed difference if Equalf assertion failed. since all the variable are mocked it is not required here. removing Equalf

File: internal/subscriber/ordering_test.go
Comment (sumitkrzp) [reply internal/subscriber/ordering_test.go]: Done

File: internal/subscriber/subscriberimpl_test.go
Comment (sakshi-goyal-razorpay) [reply internal/subscriber/subscriberimpl_test.go:256]: Increasing test coverage is not about reaching a particular no. The more coverage we have, less are the chances of committing buggy code.
Changing this health check ticker monitor has no impact on the latter, but will only increase some numbers on paper.

Comment (sumitkrzp) [reply internal/subscriber/subscriberimpl_test.go:256]: yes exactly

Comment (sumitkrzp) [reply internal/subscriber/subscriberimpl_test.go:256]: and also if there is some error in monitoring code this fail the test case

Comment (sumitkrzp) [reply internal/subscriber/subscriberimpl_test.go:256]: I don't see any downside here

Comment (sakshi-goyal-razorpay) [reply internal/subscriber/subscriberimpl_test.go:256]: @sumitkrzp Please try to figure out the reason why ticker interval change inc the code coverage in the first case, and then we will see, if it really make sense to do the change.

Comment (sumitkrzp) [reply internal/subscriber/subscriberimpl_test.go:256]: @sakshi-goyal-razorpay 
This will take care of bellow lines : 
`case <-s.healthMonitorTicker.C: 
lag := s.subscriberImpl.GetConsumerLag()
			for topic, offset := range lag {
				tp := strings.Split(topic, "=")
				if len(tp) == 1 {
					logger.Ctx(ctx).Errorw("subscriber: failed to parse topic and partition for consumer lag", "topic", topic)
					continue
				}
				subscriberPartitionConsumerLag.WithLabelValues(env, tp[0], s.subscription.Name, tp[1]).Set(float64(offset))
			}`

**Review by sakshi-goyal-razorpay — APPROVED** [review]

**Comment by sakshi-goyal-razorpay** [comment]
@sumitkrzp Akash has not reviewed the PR yet, please get his approval before merging PR

**Pull request merged**
